security-analysis/security-analysis-api/src/main/java/com/powsybl/security/detectors/AbstractContingencyBlindDetector.java

```diff
@@ -58,6 +59,21 @@ public void checkVoltage(VoltageLevel voltageLevel, Consumer<LimitViolation> con
         voltageLevel.getBusView().getBusStream().forEach(b -> checkVoltage(b, consumer));
     }
 
+    /**
+     * This implementation takes the voltageAngle difference to be checked from the Network.
+     */
+    @Override
+    public void checkVoltageAngle(VoltageAngleLimit voltageAngleLimit, Consumer<LimitViolation> consumer) {
+        Bus referenceBus = voltageAngleLimit.getTerminalFrom().getBusView().getBus();
+        Bus otherBus = voltageAngleLimit.getTerminalTo().getBusView().getBus();
+        if (referenceBus != null && otherBus != null
+            && referenceBus.getConnectedComponent().equals(otherBus.getConnectedComponent())) {
```

Comment:
should be checking if in same synchronousComponent. Angle difference for same CC but different SC is meaningless.
Also should compare component getNum() instead of using Object.equals